Controller/TwigExceptionController.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Bundle\FrameworkBundle\Templating\TemplateReference;
+use Symfony\Component\Debug\Exception\FlattenException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Log\DebugLoggerInterface;
+
+/**
+ * Custom ExceptionController that uses the view layer and supports HTTP response status code mapping.
+ * It additionally is able to prepare the template parameters for the core EngineInterface.
+ */
+class TwigExceptionController extends ExceptionController
+{
+    /**
+     * @var EngineInterface
+     */
+    private $templating;
+
+    public function setTemplating(EngineInterface $templating)
+    {
+        $this->templating = $templating;
+    }
+
+    public function getTemplating()
+    {
+        if (!$this->templating instanceof EngineInterface) {
+            throw new \RuntimeException('No templating engine set');
+        }
+
+        return $this->templating;
+    }
+
+    /**
+     * {inheritDoc}.
+     */
+    protected function createView($format, FlattenException $exception, $code, $parameters, Request $request, $showException)
+    {
+        $view = parent::createView($format, $exception, $code, $parameters, $request, $showException);
+
+        if ($this->getViewHandler()->isFormatTemplating($format)) {
+            $view->setTemplate($this->findTemplate($request, $format, $code, $showException));
+            $view->setData($parameters);
+        }
+
+        return $view;
+    }
+
+    /**
+     * {inheritDoc}.
+     */
+    protected function getParameters($currentContent, $code, $exception, DebugLoggerInterface $logger = null, $format = 'html')
+    {
+        $parameters = parent::getParameters($currentContent, $code, $exception, $logger, $format);
+
+        if ($this->getViewHandler()->isFormatTemplating($format)) {
+            $parameters['logger'] = $logger;
+        }
+
+        return $parameters;
+    }
+
+    /**
+     * Finds the template for the given format and status code.
+     *
+     * Note this method needs to be overridden in case another
+     * engine than Twig should be supported;
```

Comment:
I don't think that this is true anymore. If we want people to be able to use other templating engines, they should not extend the `TwigExceptionController`. We should then rather create a base `TemplatingExceptionController` or something like that which defines all the base logic and has an abstracht `findTemplate()` method (so you don't depend on Twig in a way when using something different).